### DISCUSSION TITLE
Gradle errors trying to build for Android

### DISCUSSION THREAD

**Discussion by jaywoodsd** [airsdk/Adobe-Runtime-Support#2820] [General]
I recently updated my AIR SDKto  50.2, and set the target Android SDK API level to 33. Now when I try to build, I keep getting the following error: "Failed to package AIR application: unexpected failure: Unable to run java: com.adobe.air.ADTException: gradle tool failed: Warning: This version only understands SDK XML versions up to 2 but an SDK XML file of version 3 was encountered. This can happen if you use versions of Android Studio and the command-line tools that were released at different times."

It seems I have the wrong gradle tools installed in my FLEX/AIR directory, but I cannot figure out how to properly overlay a new gradle version.  Has anybody run into this issue, and able to suggest how I can fix this issue? It's driving me nuts, and I'm up against a deadline for a new build.

**Comment by marchbold** [comment]
Hi,

Things to check:

- Java 11 JDK installed 
- update the Android SDK to 33 (via android studio)
- set the path to both of these in your AIR config (via the AIR SDK Manager is the easiest way)
- if you are still having issues, enable debug logging in your AIR config and post the ~/adt.log contents here.

**Reply by marchbold** [reply]
Also you can try adding the following lines to your app descriptor in the android section

```xml
    <gradleVersion>7.5</gradleVersion>
    <androidGradlePluginVersion>7.4.0</androidGradlePluginVersion>
```

**Reply by jaywoodsd** [reply]
Thank you! I followed your suggestions, and now I am able to build the APK. I think the issue is that I was using a much newer JDK (20), when I switched to JDK 11 it started working. However I have a different issue now: I am trying to build with target API level 33, to satisfy Google Play Store requirements, and even though I set targetSdkVersion="33" in my app descriptor file, the APK still targets API 31. Any idea how to make it honor the target level 33?
UPDATE: I am using AIR SDK 50.2.2 - looks like I need to upgrade to 50.2.3 to get API level 33 support.

**Reply by marchbold** [reply]
Have you placed the targetsdkversion tag in the correct position? It should be located as follows in your app descriptor:

```xml
<android>
    <manifestAdditions><![CDATA[
        <manifest android:installLocation="auto" >
            <uses-sdk android:minSdkVersion="19" android:targetSdkVersion="33" />

... 
```

Also worth ensuring you only have one `uses-sdk` node.

**Reply by jaywoodsd** [reply]
Here is a snippet from my app descriptor:
```
   <android>
        <manifestAdditions>
            <![CDATA[
		<manifest android:installLocation="auto">
		<uses-sdk android:minSdkVersion="15" />
		<uses-sdk android:targetSdkVersion="33"/>
	. . .
```
After upgrading to AIR SDK 50.2.3.5, my APK is showing targetSdkVersion=33. Note that I have 2 separate <uses-sdk> nodes... this configuration is working, but I will follow your suggestion and combine them into a single node.

Thanks again for your help! Nice to know there is still some coommunity out there for Flex/AIR developers. 
Cheers, Jay

**Reply by marchbold** [reply]
Yeah you can't have two, Android will likely pick the first so your targetsdk wouldn't have been set correctly. Glad you have it working!
